lib/hocon/impl/abstract_config_value.rb

```diff
@@ -60,6 +157,78 @@ def require_not_ignoring_fallbacks
     end
   end
 
+  def construct_delayed_merge(origin, stack)
+    # TODO: this might not work because ConfigDelayedMerge inherits
```

Comment:
The error will be like "undefined class Hocon::Impl::Whatever", with this line in the stack trace.  If that happens we will have to change the class hierarchy around significantly.

On the other hand, it might "just work" because the files that contain these classes might get `require`d from elsewhere before these lines of code are executed.  Or we could put the `require` lines here inside the method.

magic++